### PR TITLE
feat: added support for consuming new rules

### PR DESCRIPTION
**Pull request by jaig-0911**

### CODE REVIEW

File: packages/preset-rules/src/customRules.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+const customRulesPathDir = 'custom-rules';
+
+export function registerCustomRules(): void {
+    const directoryPath = path.join(__dirname, customRulesPathDir);
+    const newChecks: axe.Check[] = [];
+    const newRules: axe.Rule[] = [];
+
+    try {
+        // Path to the JSON file with changes to existing rules
+        const changesInExistingRulesPath = path.join(directoryPath, 'changes.json');
```
Comment (navateja-alagam) [inline packages/preset-rules/src/customRules.ts:21]: what does `changes.json` file nomenclature for? Does it contain customized changes to rules and checks?

File: packages/preset-rules/src/customRules.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                    targetArray.push(fileData);
+                }
+            });
+        };
+
+        // Process checks and rules JSON files
+        processJsonFiles<axe.Check>(path.join(directoryPath, 'checks'), newChecks);
+        processJsonFiles<axe.Rule>(path.join(directoryPath, 'rules'), newRules);
+    } catch (err) {
+        console.error('Error in reading Custom Rules files:', err);
```
Comment (navateja-alagam) [inline packages/preset-rules/src/customRules.ts:45]: may be better to print `err.message`

File: packages/preset-rules/src/customRules.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        const changesInExistingRulesPath = path.join(directoryPath, 'changes.json');
+        // Read and parse existing rule changes
+        const existingRuleData = fs.readFileSync(changesInExistingRulesPath, 'utf-8');
+        const { rules } = JSON.parse(existingRuleData) as { rules: axe.Rule[] };
+        if (rules && Array.isArray(rules)) {
+            newRules.push(...rules);
+        }
+
+        // Function to process JSON files in a directory and push their content to a target array
+        const processJsonFiles = <T>(dir: string, targetArray: T[]): void => {
```
Comment (navateja-alagam) [inline packages/preset-rules/src/customRules.ts:30]: for the sake of cleaner and effective coding, I would suggest to push this utility code under `@sa11y/common` more generic to accept type of file extension too

File: packages/preset-rules/src/custom-rules/changes.json [hunk 4]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (navateja-alagam) [inline packages/preset-rules/src/custom-rules/changes.json:1]: any possibility we can add something here?

File: packages/preset-rules/src/customRules.ts (hunk 1 above)
Comment (jaig-0911) [reply packages/preset-rules/src/customRules.ts:21]: changes.json nomenclature is for having any changes in the existing rules. [Reference](https://docs.deque.com/devtools-for-web/4/en/cli-rulesets#:~:text=directory%0A%20%E2%94%9C%20changes.json%0A%20%E2%94%9C%20rules%0A%20%E2%94%82%20%20%E2%94%94%20h1%2Dno%2Dduplicate.json%0A%20%E2%94%94%20checks%0A%20%20%20%20%E2%94%94%20page%2Dno%2Dduplicate%2Dh1.json)

**Review by vaibhavv06 — APPROVED** [review]
LGTM

**Review by navateja-alagam — APPROVED** [review]

**Pull request merged**